recipes/pyg-lib/build.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -ex
+
+if [[ ${cuda_compiler_version} != "None" ]]; then
+  export FORCE_CUDA=1
+  export CUDA_ARCH_LIST=75
+else
+  export FORCE_CUDA=0
+fi
+
+export USE_MKL_BLAS=1
```

Comment:
Does this force linkage against MKL?

Reply:
I am not sure how to checl that. Here is the relevant CMakeFiles piece:

```
set(WITH_MKL_BLAS 0)
if(USE_MKL_BLAS AND DEFINED BLAS_INCLUDE_DIR)
  find_file(MKL_INCLUDE_FOUND mkl.h ${BLAS_INCLUDE_DIR} NO_DEFAULT_PATH)
  if(MKL_INCLUDE_FOUND)
    set(WITH_MKL_BLAS 1)
  else()
    if(WITH_COV)
      message(FATAL_ERROR "The mkl.h file was not found - pass the correct directory or set USE_MKL_BLAS=OFF.")
    else()
      message(WARNING "The mkl.h file was not found - building pyg-lib without MKL BLAS support.")
    endif()
  endif()
endif()
```

Then `WITH_MKL_BLAS` is used within the C++ code to use or not MKL at https://github.com/pyg-team/pyg-lib/blob/0a2300ba8db276fc9140a415d4b65c9141135a8f/pyg_lib/csrc/ops/cpu/matmul_kernel.cpp#L15

Reply:
The looks very much as this would introduce a direct dependency on MKL. I'm no MKL expert but that doesn't look like it being compatible with other BLAS implementations. The build works as it seems that some other process is also pulling in MKL. 

There doesn't seem to be any direct linkage to MKL at least reported.

Reply:
> but that doesn't look like it being compatible with other BLAS implementations.

That sounds like an issue. Is it?

> There doesn't seem to be any direct linkage to MKL at least reported.

does that mean the current build is fine?

---

Should we ping someone else here?

Reply:
Yes but I'm unsure who.

Reply:
@conda-forge/blas: what do you think of the above? It seems unclear to us whether the above MKL integration is ok or not.

Reply:
I haven't looked at the recipe here in detail, but first as a general comment: we usually try to keep things compatible with several blas implementations, by ensuring we only compile against reference lapack (which can then be replaced by a given flavour in the final environment).

However, there are a handful of packages I'm aware of that hard-require mkl. I think we should strive for the former, but it's not impossible to do the latter.

Reply:
Understood and thanks for the context. According to https://github.com/pyg-team/pyg-lib/blob/fee5fa9cbd6ca559058831e61fec7ee4f32ae459/pyg_lib/csrc/ops/cpu/matmul_kernel.cpp#L16, mkl.h seems hard coded indeed in the source code. If you're fine I'll be to merge here. I can also raise a ticket upstream if you prefer and hold for it.

Reply:
Fine with merging it but then we should also list MKL as an explicit dependency.

Reply:
Ok done

Reply:
I see from the code linked above that mkl.h is conditionally included, but are you sure that `WITH_MKL_BLAS()` is evaluating to TRUE with your build? The link inspector on all platforms reports that MKL is unused (not linked dynamically).

Reply:
It's my bad here, I just realized that MKL has only been added for `>0.1.0` (likely for the coming `0.2.0`). I will simply comment the MKL relevant bits for now and uncomment them when 0.2.0 is out in the feedstock.

Reply:
Done.